arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -508,7 +520,16 @@ internal expect val IODispatcher: CoroutineDispatcher
 public suspend fun <A : AutoCloseable> ResourceScope.autoCloseable(
   closingDispatcher: CoroutineDispatcher = IODispatcher,
   autoCloseable: suspend () -> A,
-): A = install({ autoCloseable() } ) { s: A, _ -> withContext(closingDispatcher) { s.close() } }
+): A {
+  contract {
+    callsInPlace(autoCloseable, InvocationKind.EXACTLY_ONCE)
+  }
+  return withContext(NonCancellable) {
+    val s = autoCloseable()
+    onRelease { withContext(closingDispatcher) { s.close() } }
+    s
```

Comment:
This is also a change in the code. Why do we need the additional `withContext(NonCancellable)`?

Reply:
I simply inlined `install` instead of suppressing the contract warning. That's because `install` can't have a contract because it's a member function. I can undo that and instead suppress the warning if you'd like

Reply:
No, this is fine, although I would prefer if you could add some comment with the "original" code and then stating you've inline that.